src/components/macro/macro.component.ts

```diff
@@ -1,12 +1,92 @@
-import { Component } from '@angular/core';
+import { Component, OnInit, OnDestroy, ViewChildren, QueryList, AfterViewInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Subscription } from 'rxjs/Subscription';
+
+import { UhkConfigurationService } from '../../services/uhk-configuration.service';
+
+import { Macro } from '../../../config-serializer/config-items/Macro';
+import { MacroAction } from '../../../config-serializer/config-items/MacroAction';
+import { TextMacroAction } from '../../../config-serializer/config-items/TextMacroAction';
+import { MacroItemComponent } from './macro-item/macro-item.component';
+
+import { ContenteditableModel } from '../directives/contenteditable.component';
 
 @Component({
     selector: 'macro',
     template: require('./macro.component.html'),
-    styles: [require('./macro.component.scss')]
+    styles: [require('./macro.component.scss')],
+    providers: [UhkConfigurationService]
```

Comment:
Since https://github.com/UltimateHackingKeyboard/agent/commit/dee9c1077bb06014be28f09fdcb15f36f94cc884 the `UhkConfigurationService˛is provided in the app.module.ts so you don't need to provide an other.